workflow.md

```diff
@@ -100,10 +123,91 @@ Please note that actual version of project is defined by project settings,
+`vX.Y.Z` already):
+    a) `cardano-sl-X.Y-rc` should be renamed to `cardano-sl-X.Y`, `cardano-sl-X.Y-mid` immediately spawned
+    b) `cardano-sl-X.Y-mid` is immediately spawned from the same commit.
+    c) `genesis.bin`, `constants-*.yaml` should be changed.
+    d) `vX.Y.Z` should be deployed on `csl-testnet` cluster.
+    e) tada.iohk.io, daedaluswallet.io should be updated.
+
+Rules for `-rc` branch are same as for release branch, specifically:
+* `-rc` branch must be merged into `master` after every change.
+* `master` is never merged into `-rc` branch.
+* `-rc` branch must be protected.
+
+### Interaction between cardano-sl and projects dependent on it
+
+Some projects (e. g. `cardano-sl-explorer`, `daedalus`) depend on
+`cardano-sl`. In order to facilitate interaction with `cardano-sl` we
+use special branch `-mid` mentioned above. Explorer is used as an
+example in the rest of this subsection, but rules apply to everything
+dependent on `cardano-sl`.
+
+* Explorer must depend only on version of `cardano-sl` from release
+  branch or `-mid` branch, not `-rc` branch. This rule is imposed
+  because only these branches are generally stable and
+  well-tested. `-rc` branch corresponds to a version which is not
+  properly tested yet, hence explorer should not depend on it.
+* Explorer should always be launched with the same version of
+  `cardano-sl` as the version it depends on. That's because different
+  versions of `cardano-sl` may be incompatible in some way and it may
+  lead to different (inconsistent) behavior of `cardano-node` and
+  `cardano-explorer`.
+* Version `X.Y.Z` of explorer (located in `sl-explorer-X.Y` branch)
+  can use only one of `X.Y.*` versions of `cardano-sl`. Note that
+  `Fix` versions may differ.
+* If explorer developer wants to change something in `cardano-sl` for
+  explorer's version `X.Y.Z` they should make a PR to `cardano-sl-X.Y`
+  branch, wait until it's merged and then use newer version in
+  explorer.
+* `master` branch of explorer should depend on the latest `-mid`
+  branch of `cardano-sl`.
+* If explorer developers wants to change something in `cardano-sl` for
+  explorer's `master` they should make a PR to `cardano-sl-X.Y-mid` of
+  `cardano-sl` (where `X.Y` is latest version of `cardano-sl`).
+* `cardano-sl-X.Y.` should be frequently merged into
+  `cardano-sl-X.Y-mid`.
+* `cardano-sl-X.Y-mid` should be frequently merged to `master`.
+
+**TODO** Frankly, I am not sure I properly understand all this `-mid` stuff.
+Treat it as a draft which should be refined.
+**TODO** this workflow as described now is bad, because
+`sl-explorer-0.5` can be tested (and, well, can exist) only after
+`cardano-sl-0.5` is released. It's probably bad. On the other hand, we
+don't want explorer guys to use unstable (untested) version.
+**TODO** according to workflow described above there should be only
+one `-mid` branch at any time. Is it correct? So it's something like
```

Comment:
Then I need an example. Suppose there is `cardano-sl-0.8` already, what's the point of having `cardano-sl-0.5-mid` in this case, for example?

I see how `cardano-sl-0.8-mid` is useful, it can be used by latest explorer (`master`). It is basically a stable version (`cardano-sl-0.8`) + some changes from explorer guys who know what they are doing.

But I don't understand why `cardano-sl-0.5-mid` can be useful. Where is it suppose to be used? 
By explorer's release? Suppose there is explorer's release 0.2 which uses cardano-sl release 0.5. Explorer should point to some commit in `cardano-sl-0.5`. If we want to change something in explorer 0.2 which requires changes in `cardano-sl-0.5`, we should make a PR to `cardano-sl-0.8`, right? It's still obscure to me whether `-mid` branch is only for master or for release branches.
By explorer's master? If explorer lags behind master so much, then we probably want it to catch up ASAP, don't we? On the other hand, I can imagine situation that `cardano-sl-0.8` has substantial changes in API comparing to `cardano-sl-0.5` and we don't want to spend time on moving explorer to `cardano-sl-0.5`, so master is still using `-0.5`. That's the only case I see where `-0.5-mid` can be useful. Is it indeed the only case? If it is, I will write about it in `workflow.md`, because it's not obvious and it will make it easier to understand this workflow.